### PR TITLE
BE-632 | Include limit order balances in passthrough/portfolio-assets

### PR DESCRIPTION
**Pull request by deividaspetraitis**
This PR introduces changes to include limit order balances in total balance for `passthrough/portfolio-assets` endpoint. As a side effect changes includes cleaning up hardcoded order direction strings in some places.

Consider we have ask order where we sell 0.1 OSMO and want to receive 1 USD. Order is 50% filled of which 25% is already claimed, that would result in following properties of the order:

- Filled USD = `1 USD * 50%` = `1 * 0.5` = `0.5 USD`
- Filled OSMO = `0.1 OSMO * 50%` = `0.005`
- Claimed USD = `0.5 USD * 25%` = `0.5 * 0.25` = `0.125 USD`
- Remaining Claimable USD = `0.5 USD - 0.125 USD` = `0.375 USD`

Introduced changes adds `limit-orders`  section that contains unclaimed balances, as per example above such section would look like:

```json
  "limit-orders": {
     "capitalization": "0.375", // USD
     "is_best_effort": false
   },
```

Second change is that unclaimed coins will appear under `total-assets` section in `account_coins_result`, for example:

```json
   "total-assets": {
      "capitalization": "93.999502000156560410",
      "account_coins_result": [
        {
          "coin": {
            "denom": "USD",
            "amount": "0.375"
          },
          "cap_value": "0.375"
        },
      ],
      "is_best_effort": false
    },
```

At the moment I am opening this PR I am not 100% sure I have addressed the problem correctly because due following questions:

1. When I placed above order from my account disappeared 0.1 OSMO, after it's filled 50% I still own 0.05, but where it's reflected?
2. I am not sure I understand all of the sections of the output of this endpoint, for example `user-balances` vs `account_coins_result`.

## Release Notes

- **New Features**
  - Enhanced order book functionality integrated into the server, allowing for improved management of limit orders.
  - New methods added to handle order directions and calculate fillable amounts for limit orders.
  - The passthrough use case now supports fetching limit order coins alongside other asset types.

- **Bug Fixes**
  - Improved error handling in order processing methods, ensuring clearer feedback on failures.

- **Documentation**
  - Updated comments for clarity on method behaviors and structures related to order management.

### CODE REVIEW

File: passthrough/usecase/passthrough_usecase.go [hunk 1]
```diff
@@ -238,6 +245,18 @@ func (p *passthroughUseCase) GetPortfolioAssets(ctx context.Context, address str
 		return unclaimedCoins, finalErr
 	}
 
+	getLimitOrderCoins := func(ctx context.Context, address string) (sdk.Coins, error) {
+		orders, _, err := p.orderbookUseCase.GetActiveOrders(ctx, address)
+		var limitOrdersCoins sdk.Coins
+		for _, order := range orders {
+			limitOrdersCoins = limitOrdersCoins.Add(sdk.NewCoin(
```
Comment (p0mvn) [inline passthrough/usecase/passthrough_usecase.go]: Please switch to this for performance reasons. Coin constructor has extra overhead

```suggestion
			limitOrdersCoins = limitOrdersCoins.Add(sdk.Coin{
```

File: passthrough/usecase/passthrough_usecase.go [hunk 2]
```diff
@@ -238,6 +245,18 @@ func (p *passthroughUseCase) GetPortfolioAssets(ctx context.Context, address str
...
+				order.BaseAsset.Symbol,
+				order.ClaimableAmount().Ceil().TruncateInt(),
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply passthrough/usecase/passthrough_usecase.go]: Let's apply this?

File: passthrough/usecase/passthrough_usecase.go [hunk 3]
```diff
@@ -238,6 +245,18 @@
...
+			))
+		}
+		return limitOrdersCoins, err
+	}
+
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply passthrough/usecase/passthrough_usecase.go]: And this

**Comment by p0mvn** [comment]
> When I placed above order from my account disappeared 0.1 OSMO, after it's filled 50% I still own 0.05, but where it's reflected?

I don't fully understand what happened here - could you elaborate?

**Comment by p0mvn** [comment]
> I am not sure I understand all of the sections of the output of this endpoint, for example user-balances vs account_coins_result.

User balances only contain the user balances while the latter contains all of the value from pools, unclaimed rewards etc

**Comment by p0mvn** [comment]
Do you mind please updating the docs with the limit order data too please?
https://github.com/osmosis-labs/sqs/blob/f4188d5df8f2d70e4770b06e7e1b40c4237bbd0f/docs/architecture/passthrough.md#L17-L46

**Comment by p0mvn** [comment]
Let's also extend this test:
https://github.com/osmosis-labs/sqs/blob/a9e6b469aad7338f9323a19746b6b64e5140818c/tests/test_passthrough.py#L58

**Comment by deividaspetraitis** [comment]
Task got derailed to other direction.

**Pull request closed without merging**